tests/gpflux/layers/basis_functions/fourier_features/test_quadrature.py

```diff
@@ -0,0 +1,175 @@
+#
+# Copyright (c) 2021 The GPflux Contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import numpy as np
+import pytest
+import tensorflow as tf
+from tensorflow.python.keras.testing_utils import layer_test
+from tensorflow.python.keras.utils.kernelized_utils import inner_product
+
+import gpflow
+from gpflow.quadrature.gauss_hermite import NDiagGHQuadrature
+from gpflow.utilities.ops import difference_matrix
+
+from gpflux.layers.basis_functions.fourier_features import QuadratureFourierFeatures
+from gpflux.layers.basis_functions.fourier_features.utils import QFF_SUPPORTED_KERNELS
+
+
+@pytest.fixture(name="n_dims", params=[1, 2, 3])
+def _n_dims_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="n_components", params=[1, 2, 4, 20])
+def _n_features_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="variance", params=[0.5, 1.0, 2.0])
+def _variance_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="lengthscale", params=[0.75, 1.0, 5.0])
+def _lengthscale_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="batch_size", params=[1, 10])
+def _batch_size_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="kernel_cls", params=list(QFF_SUPPORTED_KERNELS))
+def _kernel_cls_fixture(request):
+    return request.param
+
+
+def test_throw_for_unsupported_kernel():
+    kernel = gpflow.kernels.Constant()
+    with pytest.raises(AssertionError) as excinfo:
+        QuadratureFourierFeatures(kernel, n_components=1)
+    assert "Unsupported Kernel" in str(excinfo.value)
+
+
+def test_quadrature_fourier_features_can_approximate_kernel_multidim(
+    kernel_cls, variance, lengthscale, n_dims
+):
+
+    n_components = 128
```

Comment:
Should we a create placeholder to a notebook explaining the lengthscale effect?
```suggestion
    """
    Compare finite feature approximation to analytical kernel expression.
    Approximation only holds for large enough lengthscales, as explained in TODO: notebook.
    """
    n_components = 128
```